core/src/main/java/org/sonarsource/sonarlint/core/websocket/SonarCloudWebSocket.java

```diff
@@ -108,12 +115,16 @@ private static Optional<? extends ServerEvent> parse(WebSocketEvent event) {
     var eventType = event.event;
     if (eventType == null) {
       return Optional.empty();
-    } else if (!parsersByType.containsKey(eventType)) {
+    } else if (!parsersByTypeForProjectFilter.containsKey(eventType) && !parsersByTypeForProjectUserFilter.containsKey(eventType)) {
       SonarLintLogger.get().error("Unknown '{}' event type ", eventType);
       return Optional.empty();
     }
     try {
-      return parsersByType.get(eventType).parse(event.data.toString());
+      var eventParser = parsersByTypeForProjectFilter.get(eventType);
+      if (eventParser == null) {
+        eventParser = parsersByTypeForProjectUserFilter.get(eventType);
+      }
+      return eventParser.parse(event.data.toString());
```

Comment:
We could have the 2 separate maps as you did, and then one field that aggregates the 2 so that it's simpler to use here